private/buf/cmd/buf/command/beta/registry/draft/draftdelete/draftdelete.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package draftdelete
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bufbuild/buf/private/buf/bufcli"
+	"github.com/bufbuild/buf/private/bufpkg/bufmodule/bufmoduleref"
+	"github.com/bufbuild/buf/private/pkg/app/appcmd"
+	"github.com/bufbuild/buf/private/pkg/app/appflag"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const forceFlagName = "force"
+
+// NewCommand returns a new Command
+func NewCommand(
+	name string,
+	builder appflag.Builder,
+) *appcmd.Command {
+	flags := newFlags()
+	return &appcmd.Command{
+		Use:   name + " <buf.build/owner/repo:draft>",
+		Short: "Delete a draft of a BSR repository by name.",
+		Args:  cobra.ExactArgs(1),
+		Run: builder.NewRunFunc(
+			func(ctx context.Context, container appflag.Container) error {
+				return run(ctx, container, flags)
+			},
+			bufcli.NewErrorInterceptor(),
+		),
+		BindFlags: flags.Bind,
+	}
+}
+
+type flags struct {
+	Force bool
+}
+
+func newFlags() *flags {
+	return &flags{}
+}
+
+func (f *flags) Bind(flagSet *pflag.FlagSet) {
+	flagSet.BoolVar(
+		&f.Force,
+		forceFlagName,
+		false,
+		"Force deletion without confirming. Use with caution.",
+	)
+}
+
+func run(
+	ctx context.Context,
+	container appflag.Container,
+	flags *flags,
+) error {
+	bufcli.WarnBetaCommand(ctx, container)
+	moduleReference, err := bufmoduleref.ModuleReferenceForString(container.Arg(0))
+	if err != nil {
+		return appcmd.NewInvalidArgumentError(err.Error())
+	}
+	if moduleReference.Reference() == bufmoduleref.Main {
+		return appcmd.NewInvalidArgumentError("you must specify a valid draft")
```

Comment:
```suggestion
		return appcmd.NewInvalidArgumentErrorf("%q is not a valid draft name", bufmoduleref.Main)
```

Reply:
updated, but thinking about this again, this one is actually a bit tricky, one of the reason of this checking is to make sure user did provide the reference, as it can be filled as `bufmoduleref.Main`  automatically by `bufmoduleref.ModuleReferenceForString`, which will led to something like this:
```
❯ buf beta registry draft delete bufbuild.internal/tommyma/test-repo   

...

Failure: "main" is not a valid draft name
```
which doesn't seems really pointing to the issue? how do you think?

Reply:
You're right that that doesn't seem right. Its unfortunate that `bufmoduleref.ModuleReferenceForString` doens't allow us to separate out the "main explicitly set" and "main implicitly inferred" case.

Differentiating the error message by `strings.HasSuffix(container.Arg(0), ":"+bufmoduleref.Main)` doesn't seem ideal either, but may be the best we can do?

Reply:
updated